### PR TITLE
go: generate and save go protobufs

### PR DESCRIPTION
**Pull request by kyessenov**
Pointing to a repository without go code is not valid go code outside of bazel environment:
```
udpa/service/orca/v1/orca.pb.go:9:2: module github.com/cncf/udpa@latest (v0.0.0-20191004202315-015fc86d90f4) found, but does not contain package github.com/cncf/udpa/udpa/data/orca/v1
```

The proposal is to use this repo to host the golang generated UDPA protos.

Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

**Comment by kyessenov** [comment]
cc @htuch @nareddyt

**Comment by htuch** [comment]
@kyessenov I'm hesitant about forming dependencies on Envoy specific projects from UDPA. Is there no way to add some Go stubs or something in this repository?

**Comment by kyessenov** [comment]
We can add them here, right next to protos. Is this something you want to commit to? It's hard to change paths once we publish the go generated code (the package prefix is very important in golang). 
We would need to set-up CI as well (I'm not seeing any CI checks).

**Comment by htuch** [comment]
@kyessenov I think we do need them here ideally (for the UDPA ones). Would you be able to do the plumbing? We also need to setup CI, @lizan do you know the best path for AZP first CI?

**Comment by kyessenov** [comment]
We also need to add go module versioning I believe and go manifests (go.mod file).

**Comment by kyessenov** [comment]
Added a script to store generated files. Checked manually with:
```
bazel test ...
go build go/...
```

We probably need a check that the repo is clean after re-generating the files.

File: go/udpa/data/orca/v1/orca_load_report.pb.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
```
Comment (htuch) [inline go/udpa/data/orca/v1/orca_load_report.pb.go]: I think GH is ignoring these, but maybe we should have something like https://github.com/envoyproxy/envoy/blob/master/.gitattributes#L2 to ensure consistent CLI and GH view of the generated files.

File: bazel/generate_go_protobuf.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    output_dir = os.path.join(output, rule_dir)
+
+    # Ensure the output directory exists
+    os.makedirs(output_dir, 0o755, exist_ok=True)
+    for generated_file in input_files:
+      output_file = shutil.copy(generated_file, output_dir)
+      os.chmod(output_file, 0o644)
+
+
+if __name__ == "__main__":
```
Comment (htuch) [inline bazel/generate_go_protobuf.py]: Who runs this? Should there be a README.md?

File: bazel/generate_go_protobuf.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
```
Comment (htuch) [inline bazel/generate_go_protobuf.py]: Prefer that this goes in `tools/`.

Comment (kyessenov) [reply bazel/generate_go_protobuf.py]: Ack, moved.

File: go/udpa/data/orca/v1/orca_load_report.pb.go (hunk 1 above)
Comment (kyessenov) [reply go/udpa/data/orca/v1/orca_load_report.pb.go]: Done.

File: bazel/generate_go_protobuf.py (hunk 2 above)
Comment (kyessenov) [reply bazel/generate_go_protobuf.py]: Added bare-bones developer README file.
It should be run by whoever wants to change the API.
It should really be a makefile, but I don't want to get into the debate of makefiles vs bazel.

File: go/go.mod [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module github.com/cncf/udpa/go
```
Comment (kyessenov) [inline go/go.mod]: NB: go module declaration cannot be placed at the root since go does not like C++ files in test/

**Pull request merged**